src/library/scala/collection/IndexedSeqView.scala

```diff
@@ -40,36 +40,67 @@ trait IndexedSeqView[+A] extends IndexedSeqOps[A, View, View[A]] with SeqView[A]
 object IndexedSeqView {
 
   @SerialVersionUID(3L)
-  private final class IndexedSeqViewIterator[A](self: IndexedSeqView[A]) extends AbstractIterator[A] with Serializable {
+  final class IndexedSeqViewIterator[A](self: IndexedSeqView[A]) extends AbstractIterator[A] with Serializable {
     private[this] var current = 0
+    private[this] var remainder = self.size
     override def knownSize: Int = self.size - current
-    def hasNext = current < self.size
+    def hasNext = remainder > 0
     def next(): A =
       if (hasNext) {
         val r = self.apply(current)
         current += 1
+        remainder -= 1
         r
       } else Iterator.empty.next()
 
     override def drop(n: Int): Iterator[A] = {
-      if (n > 0) current = Math.min(self.size, current + n)
+      if (n > 0) {
+        current = Math.min(self.size, current + n)
+        remainder = Math.max(0, remainder - n)
+      }
+      this
+    }
+
+    override def slice(from: Int, until: Int): Iterator[A] = {
```

Comment:
I think if you override `sliceIterator` rather than `slice` you'll get `slice`, and for free you'll get `take`

Reply:
`sliceIterator` is an abstraction, which adds another layer of complexity in code understanding due to dependency from different methods (`slice` and `take`) pointing to it. 
To compare with the tradeoff between reuse and clarity, I personally think overriding `slice` is better, and it's simple enough. Besides, `take` is already constant time as well in `Iterator`, and if there is a need to insist on overriding `take` on `IndexedSeqView`, it's quite simple to do as well.

Reply:
What I was thinking was that `sliceIterator` here wouldn't need to create a whole new `SliceIterator` object. It could just perform the same thing that `slice` does in your implementation, but then return `this`. Then we get the added benefit that `take` doesn't allocate. You don't think that's worth doing?

Reply:
What I meant was that the method `take` method re-using `sliceIterator` over-complicates the implementation as just a few lines will satisfy. (2 custom method calls as opposed to just 1)
But after re-considering, it appears to re-use, it will offer more benefits (another point that `take` won't allocate as you mentioned) and more worth it since it will reduce the likelihood of bugs.
Looking back, my justification for my original approach is not really that significant compared to the benefits of reusing. 
Will follow then, thanks a lot! :)